check-actions-security/action.yml

```
@@ -0,0 +1,149 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+name: |
+  Check actions security action.
+
+description: |
+  This action helps audit GitHub workflows for vulnerabilities. It finds many common
+  security issues in typical GitHub Actions CI/CD setups using
+  `zizmor <https://woodruffw.github.io/zizmor/>`_.
+
+inputs:
+
+  token:
+    description: |
+      Use the ``GH_TOKEN`` to allow online audits by zizmor.
+    required: false
+    type: string
+
+  generate-summary:
+    description: |
+      Whether to output a nicely formatted summary table showing the number
+      of security issues per workflow file. Default value is ``true``.
+    required: false
+    default: true
+    type: boolean
+
+  auditing-level:
+    description: |
+      Equivalent to 'persona' zizmor option, controls zizmor's auditing sensitivity.
+      Possible values include 'normal', 'high', and 'strict', which map to 'regular', 'pedantic',
+      and 'auditor' persona options. Default value is ``high``.
+    required: false
+    default: 'high'
+    type: string
+
+  trust-ansys-actions:
+    description: |
+      Whether to trust the ``ansys/actions``. Setting this option to true will allow pinning
+      ``ansys/actions`` with tags. This option is ignored if a custom ``zizmor.yml`` file is
+      detected in the repository. Default value is ``true``.
+    required: false
+    default: true
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: "Install uv"
+      uses: astral-sh/setup-uv@0c5e2b8115b80b4c7c5ddf6ffdd634974642d182 # v5.4.1
+
+    - name: "Install zizmor and verify installation"
+      shell: bash
+      run: |
+        uv tool install zizmor==1.6.0
+        zizmor --version
+
+    - name: "Optionally set GH_TOKEN"
+      if: ${{ inputs.token != '' }}
+      shell: bash
+      run: |  # zizmor: ignore[template-injection]
+        echo 'GH_TOKEN=${{ inputs.token }}' >> "$GITHUB_ENV"
```

```suggestion
    - name: "Optionally set GH_TOKEN"
      if: ${{ inputs.token != '' }}
      shell: bash
      env:
        TOKEN: ${{ inputs.token }}
      run: |  # zizmor: ignore[github-env]
        echo "GH_TOKEN=${TOKEN}" >> ${GITHUB_ENV}
```
Wouldn't that get rid of he template injection ?

---

Thanks for the suggestion!